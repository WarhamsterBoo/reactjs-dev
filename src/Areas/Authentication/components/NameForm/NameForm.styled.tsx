import styled from "@emotion/styled";
import { colors } from "@/styles/colors";

export const Form = styled.form`
  width: 200px;
  height: 120px;
  padding: 10px 10px 10px 10px;

  border: solid 1px black;
  background-color: ${colors.mainForm};

  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
`;
