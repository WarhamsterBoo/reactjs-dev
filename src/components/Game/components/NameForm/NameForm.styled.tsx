import styled from "@emotion/styled";

export const Form = styled.form`
  width: 200px;
  height: 120px;
  padding: 10px 10px 10px 10px;

  border: solid 1px black;
  background-color: #011627;

  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
`;
