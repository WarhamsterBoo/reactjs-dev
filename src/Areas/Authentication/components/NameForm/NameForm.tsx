import React, { FormEvent, useCallback, useState } from "react";
import { Button, InputText, Label } from "shared";
import { Form } from "./NameForm.styled";

export interface NameFormProps {
  onNameSubmit: (userName: string) => Promise<void>;
}

export const NameForm: React.FC<NameFormProps> = ({ onNameSubmit }) => {
  const [userName, setUserName] = useState<string>("");

  const onHandleInputChange = useCallback((ev: FormEvent<HTMLInputElement>) => {
    const { value } = ev.target as HTMLInputElement;
    setUserName(value);
  }, []);

  const onHandleSubmit = useCallback(
    async (ev: FormEvent) => {
      ev.preventDefault();
      await onNameSubmit(userName);
    },
    [onNameSubmit, userName]
  );

  return (
    <Form onSubmit={onHandleSubmit}>
      <Label>Hello there!</Label>
      <InputText
        required
        value={userName}
        onChange={onHandleInputChange}
        placeholder="Enter your name"
      />
      <Button type="submit">Start</Button>
    </Form>
  );
};
