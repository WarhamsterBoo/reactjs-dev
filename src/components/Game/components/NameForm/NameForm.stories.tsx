import { action } from "@storybook/addon-actions";
import React from "react";
import { NameForm } from "./NameForm";

export default { title: "Name Form Component" };

export const NameFormStory: React.FC<{}> = () => {
  return <NameForm onSubmit={action("onSubmit")} />;
};
