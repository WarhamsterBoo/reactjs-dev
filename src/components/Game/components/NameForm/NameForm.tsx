import React, { useState, useCallback, FormEvent } from "react";
import { Button, InputText } from "shared/";
import { InputsContainer } from "./NameForm.styled";

export const NameForm: React.FC<{}> = () => {
  const [userName, setUserName] = useState<string>("");

  const onHandleInputChange = useCallback((ev: FormEvent<HTMLInputElement>) => {
    const { value } = ev.target as HTMLInputElement;
    setUserName(value);
  }, []);

  return (
    <InputsContainer>
      <form>
        <fieldset>
          <legend>Hello!</legend>
          <label>
            Name:
            <InputText value={userName} onChange={onHandleInputChange} />
          </label>
          <Button>Ok</Button>
        </fieldset>
      </form>
    </InputsContainer>
  );
};
