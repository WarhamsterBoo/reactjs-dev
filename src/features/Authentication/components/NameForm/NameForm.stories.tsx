import { action } from "@storybook/addon-actions";
import React, { useState } from "react";
import { NameForm } from "./NameForm";

export default { title: "Name Form Component" };

export const NameFormStory: React.FC<{}> = () => {
  const [userName, setUserName] = useState<string>("Bob");

  return (
    <NameForm
      userName={userName}
      onUserNameChange={(userName: string) => {
        setUserName(userName);
        action("onChange")(userName);
      }}
      onNameSubmit={action("onSubmit")}
    />
  );
};
