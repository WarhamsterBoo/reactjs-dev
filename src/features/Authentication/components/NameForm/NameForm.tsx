import { Button, InputText, Label } from "@/components";
import React, { FormEvent, useCallback } from "react";
import { Form } from "./NameForm.styled";

interface NameFormProps {
  userName: string;
  onUserNameChange: (name: string) => void;
  onNameSubmit: () => void;
}

export const NameForm: React.FC<NameFormProps> = ({
  userName,
  onUserNameChange,
  onNameSubmit,
}) => {
  const onHandleInputChange = useCallback(
    (ev: FormEvent<HTMLInputElement>) => {
      const { value } = ev.target as HTMLInputElement;
      onUserNameChange(value);
    },
    [onUserNameChange]
  );

  const onHandleSubmit = useCallback(
    async (ev: FormEvent) => {
      ev.preventDefault();
      await onNameSubmit();
    },
    [onNameSubmit]
  );

  return (
    <Form onSubmit={onHandleSubmit}>
      <Label>Hello there!</Label>
      <InputText
        required
        value={userName}
        onChange={onHandleInputChange}
        placeholder="Enter your name"
      />
      <Button type="submit">Start</Button>
    </Form>
  );
};
