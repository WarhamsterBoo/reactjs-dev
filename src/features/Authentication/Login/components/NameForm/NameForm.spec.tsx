import { mount, shallow } from "enzyme";
import React from "react";
import { NameForm } from "./NameForm";

describe("Name Form", () => {
  it("should render", () => {
    const sut = shallow(
      <NameForm
        userName={"Bob"}
        onUserNameChange={jest.fn()}
        onNameSubmit={jest.fn()}
      />
    );

    expect(sut).toMatchSnapshot();
  });

  it("should call onLoginNameChange when loginName changes in input", () => {
    const onLoginNameChanges = jest.fn();
    const sut = mount(
      <NameForm
        userName={"Bob"}
        onUserNameChange={onLoginNameChanges}
        onNameSubmit={jest.fn()}
      />
    );

    sut
      .find(`input[type="text"]`)
      .at(0)
      .simulate("change", {
        target: { value: "John Doe", name: "name" },
      });

    expect(onLoginNameChanges).toHaveBeenCalledTimes(1);
    expect(onLoginNameChanges).toHaveBeenCalledWith("John Doe");
  });

  it("should call onNameSubmit when submit button clicked", () => {
    const fakeOnSubmit = jest.fn();
    const sut = mount(
      <NameForm
        userName={"Bob"}
        onUserNameChange={jest.fn()}
        onNameSubmit={fakeOnSubmit}
      />
    );
    sut.find(`input[type="text"]`).simulate("change", {
      target: { value: "Jane Doe" },
    });

    sut.find("button").simulate("submit");

    expect(fakeOnSubmit).toHaveBeenCalledTimes(1);
  });
});
