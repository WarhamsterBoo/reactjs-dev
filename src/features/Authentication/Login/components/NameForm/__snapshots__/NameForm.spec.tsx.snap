// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name Form should render 1`] = `
<Styled(form)
  onSubmit={[Function]}
>
  <Styled(label)>
    Hello there!
  </Styled(label)>
  <InputText
    onChange={[Function]}
    placeholder="Enter your name"
    required={true}
    value="Bob"
  />
  <Styled(button)
    type="submit"
  >
    Start
  </Styled(button)>
</Styled(form)>
`;
